Added Server.RegisterValidateRequestFunc

colega opened the pull request:
When using struct validation like github.com/asaskevich/govalidator
your service methods always start with same code:

    if err := govalidator.ValidateStruct(req); err != nil {
        return err
    }

This commit introduces a new callback that receives an already
unmarshalled request, and can interrupt the request by returning an
error.

colega committed:
Added Server.RegisterValidateRequestFunc

When using struct validation like github.com/asaskevich/govalidator
your service methods always start with same code:

    if err := govalidator.ValidateStruct(req); err != nil {
        return err
    }

This commit introduces a new callback that receives an already
unmarshalled request, and can interrupt the request by returning an
error.

tcard commented on v2/server.go:
```diff
@@ -91,6 +92,15 @@ func (s *Server) RegisterBeforeFunc(f func(i *RequestInfo)) {
+// won't be invoked and this error will be considered as the method result.
+// The argument of this function is the already-unmarshalled *args parameter of the method.
+func (s *Server) RegisterValidateRequestFunc(f func(i interface{}) error) {
+	s.validateFunc = f
```
If you reflect.ValueOf this here, then below you don't have to get out of Reflectworld and in again to call this function.

colega replied:
Good idea, thanks!

colega committed:
Moved ValidationFunc to the reflected world

So we don't have to bring stuff from reflected world, to call it, and
then reflect them again.
